File: app/src/main/java/pro/sketchware/dialogs/ProgressDialog.java
package pro.sketchware.dialogs;

import android.content.Context;
import android.content.DialogInterface;
import android.view.LayoutInflater;

import androidx.annotation.StringRes;
import androidx.appcompat.app.AlertDialog;

import com.google.android.material.dialog.MaterialAlertDialogBuilder;

import a.a.a.xB;
import pro.sketchware.R;
import pro.sketchware.databinding.ProgressMsgBoxBinding;

public class ProgressDialog {

    private final Context context;
    private final ProgressMsgBoxBinding binding;
    private boolean isCancelable = true;

    private AlertDialog dialog;

    public ProgressDialog(Context context) {
        this.context = context;
        binding = ProgressMsgBoxBinding.inflate(LayoutInflater.from(context));
    }

    public void setMessage(String message) {
        binding.tvProgress.setText(message);
    }

    public void setOnCancelListener(DialogInterface.OnCancelListener onCancelListener) {
        if (dialog != null) dialog.setOnCancelListener(onCancelListener);
    }

    public void show() {
        if (dialog == null) {
            var dialogBuilder = new MaterialAlertDialogBuilder(context);
            dialogBuilder.setTitle(getStr(R.string.common_message_loading));
            dialogBuilder.setView(binding.getRoot());
            dialog = dialogBuilder.create();
            dialog.setCancelable(false);
            dialog.setCanceledOnTouchOutside(false);
        }
        dialog.show();
    }

    public void dismiss() {
        if (dialog != null && dialog.isShowing()) dialog.dismiss();
    }

    public boolean isShowing() {
        return dialog != null && dialog.isShowing();
    }

    public boolean isCancelable() {
        return isCancelable;
    }

    public void setCancelable(boolean isCancelable) {
        this.isCancelable = isCancelable;
    }

    private String getStr(@StringRes int resId) {
        return xB.b().a(context, resId);
    }
}

File: app/src/main/java/a/a/a/ProgressDialog.kt
package a.a.a

import android.app.Dialog
import android.content.Context
import android.view.View
import android.view.Window
import android.widget.TextView
import com.sketchware.remod.R

class ProgressDialog(var1: Context) : Dialog(var1) {
    private val tvProgress: TextView
    private var isCancelable = false

    init {
        window?.requestFeature(Window.FEATURE_NO_TITLE)
        setContentView(R.layout.progress_msg_box)
        setTitle(xB.b().a(var1, R.string.common_message_progress))
        tvProgress = findViewById<View>(R.id.tv_progress) as TextView
        tvProgress.text = xB.b().a(var1, R.string.common_message_loading)
        super.setCanceledOnTouchOutside(false)
        super.setCancelable(true)
    }

    fun setProgress(progress: String?) {
        tvProgress.text = progress
    }

    fun setIsCancelable(cancelable: Boolean) {
        isCancelable = cancelable
    }

    fun isDialogCancelable(): Boolean {
        return isCancelable
    }

    override fun onBackPressed() {
        if (!isCancelable) {
            super.onBackPressed()
        }
    }
}
